### PR TITLE
feat(entity): implement removeAttachment(permission)

### PR DESCRIPTION
**Pull request by Silthus**
# Description
Permissible EntityMock now implements removeAttachment to remove permissions from a mock player.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Version number updated in `settings.gradle`.
- [x] Unit tests added.
- [x] Code follows existing code format.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java [hunk 1]
```diff
@@ -376,9 +377,22 @@ public PermissionAttachment addAttachment(Plugin plugin, int ticks)
...
+			PermissionRemovedExecutor ex = attachment.getRemovalCallback();
 
+			if (ex != null) {
+				ex.attachmentRemoved(attachment);
+			}
+
+			recalculatePermissions();
+		} else {
+			throw new IllegalArgumentException("Given attachment is not part of Permissible object " + this);
+		}
```
Comment (seeseemelk) [inline src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java]: Please use Allman braces (opening and closing brace always one a separate line).

**Review by seeseemelk — APPROVED** [review]
Looks good, except for a small style note! Thanks for the PR!

**Comment by Silthus** [comment]
Ok I fixed the style.

On a small side node: are you aware of [semantic-release](https://github.com/semantic-release/semantic-release). It could help you in keeping the version numbers up-to-date and remove the manual version bumping task.

You can see an example here: https://github.com/Silthus/spigot-plugin-template/blob/master/package.json

**Comment by Silthus** [comment]
Oh damn, I did not see that this PR is still open and added another fix and feature for commands. Hope this is ok.

Here are the changes:

- fixed an issue where dynamically registered commands could not be executed.
- added the `getHelpMap()` implementation

**Review by seeseemelk — APPROVED** [review]

**Pull request merged**